go-controller/pkg/util/nicstobridge.go

```diff
@@ -224,10 +223,6 @@ func NicToBridge(iface string) (string, error) {
 
 	// Get ip addresses and routes before any real operations.
 	family := syscall.AF_UNSPEC
-	// FIXME: Should move all interfaces over, but in IPv4 only, breaks.
```

Comment:
So this code is hit if you deploy ovn-k without the shared gw bridge already created. A user may deploy ovn-k without KIND, where they might not have already provisioned their shared gw bridge (and be running ipv4 only) and would hit this. I think the proper fix is to detect if the link has the family, then move the address over? I'm not sure what "breaks" about the current code, other than it might be trying to move the ipv6 link local address?

Reply:
Docker only enables IPv6 in the eth0 interface and disables it globally.
The code breaks because eth0 has a global ipv6 address, and when it tries to create in the breth0 interface, it fails because it has ipv6 disable.
I´m mostly sure that it doesn't try to move the link-local only global addresses.
An user with IPv4 only deployments will now break

Reply:
ah I see so this only comes into play when you have:
sysctl net.ipv6.conf.all.disable_ipv6 set to 1, which shouldn't be the case in a normal deployment. Thanks for the clarification.

Reply:
It's even worse than that; it only comes into play when you have net.ipv6.conf.all.disable_ipv6 set to 1, but somehow you also have managed to get an IPv6 address onto eth0... You pretty much have to be _trying_ to lose.